grpcbp/client_middlewares.go

```diff
@@ -101,3 +105,87 @@ func ForwardEdgeContextStreaming(ecImpl ecinterface.Interface) grpc.StreamClient
 		return nil, errors.New("grpc.ForwardEdgeContextStreaming: not implemented")
 	}
 }
+
+// PrometheusUnaryClientInterceptor is a client-side interceptor that provides Prometheus
+// monitoring for Unary RPCs.
+//
+// It emits the following metrics:
+//
+// * grpc_client_active_requests gauge with labels:
+//
+//   - grpc_service: the fully qualified name of the gRPC service
+//   - grpc_method: the name of the method called on the gRPC service
+//   - grpc_slug: an arbitray short string representing the backend the client is connecting to, the serverSlug arg
+//
+// * grpc_client_latency_seconds histogram and grpc_client_requests_total
+//   counter with labels:
+//
+//   - grpc_service, grpc_method, grpc_slug
+//   - grpc_success: "true" if status is OK, "false" otherwise
+//   - grpc_type: type of request, i.e unary
+//   - grpc_code: the human-readable status code, e.g. OK, Internal, etc
+func PrometheusUnaryClientInterceptor(serverSlug string) grpc.UnaryClientInterceptor {
+	return func(
+		ctx context.Context,
+		method string,
+		req interface{},
+		reply interface{},
+		cc *grpc.ClientConn,
+		invoker grpc.UnaryInvoker,
+		opts ...grpc.CallOption,
+	) (err error) {
+		start := time.Now()
+		serviceName, m := serviceAndMethodSlug(method)
```

Comment:
sorry for missing this; I don't really like one-letter variable names unless I could tell you what it meant with no context (i, w, r, etc.)... so maybe name this "method" and call method "serviceMethod" or "wireMethod"?